filters/include/pcl/filters/impl/farthest_point_sampling.hpp

```diff
@@ -0,0 +1,107 @@
+/*
+ * Software License Agreement (BSD License)
+ *
+ *  Point Cloud Library (PCL) - www.pointclouds.org
+ *  Copyright (c) 2009, Willow Garage, Inc.
+ *  Copyright (c) 2012-, Open Perception, Inc.
+ *  Copyright (c) 2015, Google, Inc.
+ *
+ *  All rights reserved.
+ *
+ *  Redistribution and use in source and binary forms, with or without
+ *  modification, are permitted provided that the following conditions
+ *  are met:
+ *
+ *   * Redistributions of source code must retain the above copyright
+ *     notice, this list of conditions and the following disclaimer.
+ *   * Redistributions in binary form must reproduce the above
+ *     copyright notice, this list of conditions and the following
+ *     disclaimer in the documentation and/or other materials provided
+ *     with the distribution.
+ *   * Neither the name of the copyright holder(s) nor the names of its
+ *     contributors may be used to endorse or promote products derived
+ *     from this software without specific prior written permission.
+ *
+ *  THIS SOFTWARE IS PROVIDED BY THE COPYRIGHT HOLDERS AND CONTRIBUTORS
+ *  "AS IS" AND ANY EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT
+ *  LIMITED TO, THE IMPLIED WARRANTIES OF MERCHANTABILITY AND FITNESS
+ *  FOR A PARTICULAR PURPOSE ARE DISCLAIMED. IN NO EVENT SHALL THE
+ *  COPYRIGHT OWNER OR CONTRIBUTORS BE LIABLE FOR ANY DIRECT, INDIRECT,
+ *  INCIDENTAL, SPECIAL, EXEMPLARY, OR CONSEQUENTIAL DAMAGES (INCLUDING,
+ *  BUT NOT LIMITED TO, PROCUREMENT OF SUBSTITUTE GOODS OR SERVICES;
+ *  LOSS OF USE, DATA, OR PROFITS; OR BUSINESS INTERRUPTION) HOWEVER
+ *  CAUSED AND ON ANY THEORY OF LIABILITY, WHETHER IN CONTRACT, STRICT
+ *  LIABILITY, OR TORT (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN
+ *  ANY WAY OUT OF THE USE OF THIS SOFTWARE, EVEN IF ADVISED OF THE
+ *  POSSIBILITY OF SUCH DAMAGE.
+ *
+ */
+
+#ifndef PCL_FILTERS_FARTHEST_POINT_SAMPLING_IMPL_H_
+#define PCL_FILTERS_FARTHEST_POINT_SAMPLING_IMPL_H_
+
+#include <limits>
+#include <algorithm>
+#include <pcl/filters/farthest_point_sampling.h>
+#include <pcl/common/geometry.h>
+#include <pcl/point_cloud.h>
+#include <pcl/point_types.h>
+
+
+template<typename PointT> void
+pcl::FarthestPointSampling<PointT>::applyFilter (PointCloud &output)
+{
+    int initial_size = input_->points.size();
+    int sample_size = sample_;
+    int max_index;
+    std::vector<int> selected_indices;
+    std::vector<float>distances_to_selected_points;
+    
+    //set random seed
+    std::srand (seed_);
+
+    
+    //check if number of points in cloud is less than requested number of points
+    if (initial_size <= sample_size)
+    {
+        PCL_THROW_EXCEPTION (BadArgumentException,
+                           "Requested number of points is greater than point cloud size!");
+        return;
+    }
+
+    //set initial distances as infinite
+    for (int m = 0; m < initial_size; m++)
+      distances_to_selected_points.push_back(std::numeric_limits<float>::infinity());
```

Comment:
I'm not quite sure I follow. Wouldn't I need to know the size of the point cloud to create a vector of known size in the ctor? Since the size is only known when filter.setInputCloud() is called, how could I get the size in the ctor?

Reply:
Would `max` be better compared to `infinity`? The only reason to use `infinity` would be that some actual point might overflow in distance computation to `max` and not overflow to `infinity`. Using `max` would be faster on systems with simulated `float`

Reply:
I don't suppose a distance calculated between two points would overflow under normal circumstances would it? And if it does, then this whole algorithm would fail anyway. So if max is faster, i reckon it makes sense to switch to max.

Reply:
Just a note to be clear: `max` is only faster on hardware without dedicted FP units